### PR TITLE
Create password (required step)

### PR DESCRIPTION
**Pull request by rickycodes**
**Wallet Setup:**

image

**Create password:**

image

**Create password (filled):**

image

**TODO:**

- [x] I think I still need to tweak vault encryption as I still need to use the password for that.

- [x] Also the Create password previously had a switch for using biometrics, but that doesn't seem present in the new design... pretty sure we still need it?

- [x] The logo is off centre on the Create password screen, I need to fix that. :thinking:

### CODE REVIEW

**Comment by rickycodes** [comment]
I think it's worth noting that the stepper graphic kinda falls apart on smaller screens:

image

File: app/components/Nav/App/index.js [hunk 1]
```diff
@@ -39,6 +40,9 @@ const OnboardingNav = createStackNavigator(
 		CreateWallet: {
 			screen: CreateWallet
 		},
+		ChoosePassword: {
+			screen: ChoosePassword
```
Comment (estebanmino) [inline app/components/Nav/App/index.js:44]: oh wow we didn't have this screen on the nav stack?

File: app/components/UI/OnboardingProgress/index.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+export default class OnboardingProgress extends PureComponent {
+	ranOnce = false;
+	marker = undefined;
+	state = {
+		offset: 0
+	};
+
+	static propTypes = {
+		currentStep: PropTypes.number.isRequired,
+		steps: PropTypes.array.isRequired
```
Comment (estebanmino) [inline app/components/UI/OnboardingProgress/index.js:86]: can you add small descriptions here?

File: app/components/Views/ChoosePassword/index.js [hunk 3]
```diff
@@ -215,7 +237,22 @@ class ChoosePassword extends PureComponent {
...
+				await Engine.resetState();
+				await KeyringController.createNewVaultAndKeychain(password);
+				this.keyringControllerPasswordSet = true;
+				await SecureKeychain.setGenericPassword('metamask-user', password);
+				await AsyncStorage.removeItem('@MetaMask:biometryChoice');
+				await AsyncStorage.removeItem('@MetaMask:nextMakerReminder');
+				await AsyncStorage.setItem('@MetaMask:existingUser', 'true');
+			} else {
+				// else recreateVault with password like we did previously
+				await this.recreateVault(password);
```
Comment (estebanmino) [inline app/components/Views/ChoosePassword/index.js:254]: 👌

File: app/components/Views/ChoosePassword/index.js [hunk 4]
```diff
@@ -366,33 +379,36 @@ class ChoosePassword extends PureComponent {
...
-					onValueChange={rememberMe => this.setState({ rememberMe })} // eslint-disable-line react/jsx-no-bind
-					value={this.state.rememberMe}
-					style={styles.biometrySwitch}
-					trackColor={Device.isIos() ? { true: colors.green300, false: colors.grey300 } : null}
-					ios_backgroundColor={colors.grey300}
-				/>
+				{biometryType ? (
+					<>
+						<Text style={styles.biometryLabel}>
+							{strings(`biometrics.enable_${biometryType.toLowerCase()}`)}
```
Comment (estebanmino) [inline app/components/Views/ChoosePassword/index.js:388]: is it possible to save the state in lowercase? can it be null if not set? https://github.com/MetaMask/metamask-mobile/pull/1622/files#diff-05500d00f4a6959745b57a220987a4b6R197

**Review by estebanmino — APPROVED** [review]
added very small comments, LGTM

**Pull request merged**